web/src/settings.ts

```diff
@@ -40,6 +40,31 @@ function user_can_change_password(): boolean {
     return realm.realm_email_auth_enabled;
 }
 
+function user_can_change_role(): boolean {
+    if (current_user.is_owner && !people.is_current_user_only_owner()) {
+        return true;
+    } else if (current_user.is_admin && !current_user.is_owner) {
+        return true;
+    }
+    return false;
```

Comment:
I think this can be simplified as -
```js
function user_can_change_role(): boolean {
    if (people.is_current_user_only_owner()) {
        return false;
    }
    return current_user.is_admin;
}
```

Reply:
Also, this function should be moved to `settings_data` where we have other similar functions and we can probably name this function as `user_can_change_their_own_role`.